sdk/src/main/java/io/opentelemetry/sdk/trace/SpanBuilderSdk.java

```diff
@@ -215,17 +215,28 @@ private static TimestampConverter getTimestampConverter(Span parent) {
   private static SpanContext parent(
```

Comment:
I am ok with the change overall, I kind of don't like the complicated logic in the parent function.

**Option 1**

I think this fix should be much simpler. At line 163:
```java
  if (parentContext == null || !parentContext.isValid()) {
    // New root span.
    traceId = generateRandomTraceId(random);
    // This is a root span so no remote or local parent.
    parentContext = null;
  } 
```

Then all the changes in the parent function can be dropped.

**Option 2**

Another possibility is to change where we return a SpanContext and put that check to return null if invalid.

Reply:
I'd have preferred to do all the checks in the `parent()` method, but lets follow Option 1 ;) So this is updated. Let me know if we can merge it, as I need it to update my long pending testbed PR ;)